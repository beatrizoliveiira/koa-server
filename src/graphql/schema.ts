import { buildSchema } from 'graphql';

export default buildSchema(`
  type Event {
    id: ID
    title: String
    allDay: Boolean
    start: String
    end: String
  }
  type Query {
    listEvents: [Event]
  }
  type Mutation {
    createEvent(
      title: String!,
      start: String, 
      end: String, 
      allDay: Boolean): Event
  }
`);