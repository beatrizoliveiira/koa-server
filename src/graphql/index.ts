export { registerLoader, getDataloaders } from "./loaderRegister";
export { connectionArgs, connectionDefinitions } from "./connectionDefinitions";