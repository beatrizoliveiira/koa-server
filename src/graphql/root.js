const Event = require('../models/event');

module.exports = {
  listEvents() {
    return Event.find();
  },
  createEvent(input) {
    const newEvent = new Event(input);
    return newEvent.save();
  },
}