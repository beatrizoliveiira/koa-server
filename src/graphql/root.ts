import Event from '../models/event';

export function listEvents() {
  return Event.find();
}
export function createEvent(input) {
  const newEvent = new Event(input);
  return newEvent.save();
}