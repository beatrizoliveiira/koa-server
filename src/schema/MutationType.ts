import { GraphQLObjectType } from 'graphql';

import EventMutations from '../modules/event/mutations';

const MutationType = new GraphQLObjectType({
  name: 'Mutation',
  fields: () => ({
  ...EventMutations,
  }),
});

export default MutationType;