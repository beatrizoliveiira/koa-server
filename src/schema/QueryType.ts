import { GraphQLObjectType, GraphQLNonNull } from 'graphql';

import { connectionArgs } from "../graphql/connectionDefinitions";

import { nodesField, nodeField } from '../modules/node/typeRegister';
import * as EventLoader from '../modules/event/EventLoader';
import { EventConnection } from '../modules/event/EventType';

const QueryType = new GraphQLObjectType({
  name: 'Query',
  description: 'The root of all... queries',
  fields: () => ({
    node: nodeField,
    nodes: nodesField,
    events: {
      type: new GraphQLNonNull(EventConnection.connectionType),
      args: {
        ...connectionArgs,
      },
      resolve: async (_, args, context) => await EventLoader.loadAll(context, args),
    },
  }),
});

export default QueryType;
