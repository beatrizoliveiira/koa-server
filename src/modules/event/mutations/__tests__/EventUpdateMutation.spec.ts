import { graphql } from 'graphql';
import {
  clearDbAndRestartCounters,
  connectMongoose,
  disconnectMongoose,
  sanitizeTestObject,
} from '../../../../../test';

import { createEvent } from '../../fixture/createEvent';

import { getDataloaders } from '../../../../graphql/loaderRegister';

import { schema } from '../../../../schema/schema';
import { toGlobalId } from 'graphql-relay';

beforeAll(connectMongoose);

beforeEach(clearDbAndRestartCounters);

afterAll(disconnectMongoose);

it('should update a event', async () => {
  const event = await createEvent({
    name: 'event A',
    start: '2022-01-01T00:00:00.000Z',
    end: '2022-01-01T23:59:59.000Z',
    allDay: true,
  });

  const query = `
    mutation M ($eventId: ID!, $name: String!, $start: String!, $end: String!, $allDay: Boolean!) {
      EventUpdate (input: { eventId: $eventId, name: $name, start: $start, end: $end, allDay: $allDay }) {
        event {
          id
          name
          start
          end
          allDay
        }
        error
        success
      }
    }
  `;

  const variables = {
    eventId: toGlobalId("Event", event._id),
    name: 'event B',
    start: '2022-01-02T00:00:00.000Z',
    end: '2022-01-02T23:59:59.000Z',
    allDay: false,
  };

  const rootValue = {};

  const contextValue = { dataloaders: getDataloaders() };

  const result = await graphql(schema, query, rootValue, contextValue, variables);

  console.log(result.data.EventUpdate);

  expect(result.errors).toBeUndefined();
  expect(result.data.EventUpdate.error).toBeNull();
  
  expect(result.data.EventUpdate.success).toBe('Event updated /o/');
  
  expect(result.data.EventUpdate.event.name).toBe('event B');
  expect(result.data.EventUpdate.event.start).toBe('2022-01-02T00:00:00.000Z');
  expect(result.data.EventUpdate.event.end).toBe('2022-01-02T23:59:59.000Z');
  expect(result.data.EventUpdate.event.allDay).toBe(false);

  expect(sanitizeTestObject(result.data)).toMatchSnapshot();
});