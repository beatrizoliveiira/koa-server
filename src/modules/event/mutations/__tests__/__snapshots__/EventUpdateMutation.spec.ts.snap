// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should update a event 1`] = `
Object {
  "EventUpdate": Object {
    "error": "EMPTY",
    "event": Object {
      "allDay": false,
      "end": "2022-01-02T23:59:59.000Z",
      "id": "FROZEN-ID",
      "name": "event B",
      "start": "2022-01-02T00:00:00.000Z",
    },
    "success": "Event updated /o/",
  },
}
`;
